Taskfile.yaml

```diff
@@ -0,0 +1,224 @@
+    dir: '{{.CLOUD_PROVIDER}}'
+    env:
+      AWS_ACCESS_KEY_ID: '{{.DA_AWS_ACCESS_KEY_ID}}'
+      AWS_SECRET_ACCESS_KEY: '{{.DA_AWS_SECRET_ACCESS_KEY}}'
+      AWS_DEFAULT_REGION: '{{ default "us-west-2" .AWS_DEFAULT_REGION}}'
+    cmds:
+      - |
+        terraform destroy -auto-approve {{.TF_SSH_USER_VAR}} \
+        -var='cloud_provider={{.CLOUD_PROVIDER}}' \
+        -var='deployment_prefix={{.DEPLOYMENT_ID}}' \
+        -var='public_key_path={{.PUBLIC_KEY}}' \
+        -var='nodes={{.NUM_NODES}}' \
+        {{.CLI_ARGS}}
+
+  install-role-requirements:
+    desc: install required ansible roles
+    env:
+      ANSIBLE_ROLES_PATH: '{{.ARTIFACT_DIR}}/roles'
+    cmds:
+      - mkdir -p "$ANSIBLE_ROLES_PATH"
+      - ansible-galaxy install -r "ansible/requirements.yml"
+
+  basic:
+    desc: run an ansible playbook to create a basic private cluster
+    summary: |
+      You will need to export PRIV_KEY_LOC with the location of the private key matching the public key you provided 
+      in the terraform apply task for the ansible playbook to successfully connect and apply.
+    deps:
+      - install-role-requirements
+    run: always
+    vars:
+      ANSIBLE_PLAYBOOK: '{{default "provision-node.yml" .ANSIBLE_PLAYBOOK}}'
+    env:
+      ANSIBLE_LOG_PATH: '{{.ARTIFACT_DIR}}/logs/{{.CLOUD_PROVIDER}}_{{.DEPLOYMENT_ID}}.log'
+      ANSIBLE_INVENTORY: '{{.ARTIFACT_DIR}}/hosts_{{.CLOUD_PROVIDER}}_{{.DEPLOYMENT_ID}}.ini'
+      ANSIBLE_ROLES_PATH: '{{.ARTIFACT_DIR}}/roles'
+    cmds:
+      - mkdir -p {{.ARTIFACT_DIR}}/logs
+      - ansible-playbook {{.CLI_ARGS}} ansible/playbooks/{{.ANSIBLE_PLAYBOOK}} --private-key {{.PRIVATE_KEY}}
+
+  create-ca:
+    desc: creates a local certificate authority for issuing certs
+    summary: |
+      Run if you'd like to test the TLS cluster without having your own certs on hand. Should only need to be run once
+      but if you delete it will need to be run again. It is technically a soft dependecy of the tls command
+    deps:
+      - install-role-requirements
+    run: always
+    env:
+      ANSIBLE_LOG_PATH: '{{.ARTIFACT_DIR}}/logs/{{.CLOUD_PROVIDER}}_{{.DEPLOYMENT_ID}}.log'
+      ANSIBLE_INVENTORY: '{{.ARTIFACT_DIR}}/hosts_{{.CLOUD_PROVIDER}}_{{.DEPLOYMENT_ID}}.ini'
+      ANSIBLE_ROLES_PATH: '{{.ARTIFACT_DIR}}/roles'
+    cmds:
+      - ansible-playbook {{.CLI_ARGS}} ansible/playbooks/create-ca.yml
+
+  # TODO this will need an input variable with a default setting for the hosts file variables but is fine static for now
+  populate-hosts:
+    desc: populate hosts file for a public tls cluster
+    run: always
+    vars:
+      ANSIBLE_INVENTORY: '{{.ARTIFACT_DIR}}/hosts_{{.CLOUD_PROVIDER}}_{{.DEPLOYMENT_ID}}.ini'
+    cmds:
+      - |
+        cat <(echo '[all:vars]
+        tls=true
+        advertise_public_ips=true
+
+        '
+        ) {{.ANSIBLE_INVENTORY}} > tmp.ini
+        rm {{.ANSIBLE_INVENTORY}}
+        mv tmp.ini {{.ANSIBLE_INVENTORY}}
```

Comment:
the terraform output already generates this, would it be better to generate it via `terraform output` instead?

Reply:
So the disadvantage of doing it that way is we can't support BYOB hosts files which I wanted to do on the assumption that some people might use this workflow while not using the terraform side (even if this really only exists for CI and developer testing). 

I may update the terraform output to support different Redpanda vars that ansible can handle in the future so that they flow together better but its a low prio.